liquibase-bigquery/src/main/java/liquibase/snapshot/jvm/BigQueryForeignKeySnapshotGenerator.java

```diff
@@ -0,0 +1,133 @@
+            return PRIORITY_NONE;
+        }
+    }
+
+    @Override
+    public Class<? extends SnapshotGenerator>[] replaces() {
+        return new Class[]{ForeignKeySnapshotGenerator.class};
+    }
+
+    @Override
+    protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot snapshot) throws DatabaseException, InvalidExampleException {
+        Database database = snapshot.getDatabase();
+        Table fkTable = ((ForeignKey) example).getForeignKeyTable();
+        String searchTableName = database.correctObjectName(fkTable.getName(), Table.class);
+        String fkFullName = searchTableName + "." + example.getName();
+        String searchCatalog = ((AbstractJdbcDatabase) database).getJdbcCatalogName(fkTable.getSchema());
+        String searchSchema = ((AbstractJdbcDatabase) database).getJdbcSchemaName(fkTable.getSchema());
+        String systemSchema = database.getSystemSchema().toUpperCase();
+
+        String query = new StringBuilder("SELECT ")
+                .append("TC.CONSTRAINT_NAME as CONSTRAINT_NAME, ")
+                .append("KCU.TABLE_CATALOG as FOREIGN_KEY_TABLE_CATALOG, ")
+                .append("KCU.TABLE_SCHEMA as FOREIGN_KEY_TABLE_SCHEMA, ")
+                .append("KCU.TABLE_NAME as FOREIGN_KEY_TABLE, ")
+                .append("KCU.COLUMN_NAME as FOREIGN_KEY_COLUMN, ")
+                .append("CCU.TABLE_CATALOG as PRIMARY_KEY_TABLE_CATALOG, ")
+                .append("CCU.TABLE_SCHEMA as PRIMARY_KEY_TABLE_SCHEMA, ")
+                .append("CCU.TABLE_NAME as PRIMARY_KEY_TABLE, ")
+                .append("CCU.COLUMN_NAME as PRIMARY_KEY_COLUMN ")
+                .append(String.format("FROM %1$s.%2$s.TABLE_CONSTRAINTS as TC JOIN %1$s.%2$s.CONSTRAINT_COLUMN_USAGE as CCU on " +
+                                "TC.CONSTRAINT_NAME=CCU.CONSTRAINT_NAME JOIN %1$s.%2$s.KEY_COLUMN_USAGE as KCU on KCU.CONSTRAINT_NAME=TC.CONSTRAINT_NAME ",
+                        searchSchema, systemSchema))
+                .append(String.format("WHERE TC.TABLE_NAME='%s' AND TC.table_schema='%s' AND TC.table_catalog='%s' AND TC" +
+                        ".constraint_type='FOREIGN KEY' AND TC.CONSTRAINT_NAME='%s'", searchTableName, searchSchema, searchCatalog, fkFullName))
+                .toString();
+        List<Map<String, ?>> results = Scope.getCurrentScope().getSingleton(ExecutorService.class)
+                .getExecutor("jdbc", database).queryForList(new RawSqlStatement(query));
+
+        if (!results.isEmpty()) {
+            ForeignKey foreignKey = null;
+            for (Map<String, ?> resultMap : results) {
+                Table foreignKeyTable = new Table().setName(Objects.toString(resultMap.get("FOREIGN_KEY_TABLE")));
+
+                String foreignKeyName =
+                        Optional.ofNullable((String) resultMap.get("CONSTRAINT_NAME"))
+                                .map(s -> s.replace(foreignKeyTable.getName() + ".", ""))
+                                .orElse("null");
```

Comment:
Should it be `null`? Not a String `"null"`.

Reply:
if we found FK constraint but it doesn't have a name, (what is unlikely in the first place) we can put it's name as 'null' string or null. 
Null can cause NPE, 'null' should just appear in shapshot as the FK name. 
@filipelautert what's your opinion?

Reply:
Core doesn't account for this case - it always expect a fk name to be found. So maybe something like "undefined" would be better, as "null" may confuse people. What do think?